aws-sdk-cpp: remove flaky test

happysalada opened the pull request:
###### Motivation for this change

removing another flaky test on aws-sdk-cpp https://hydra.nixos.org/build/150722293/nixlog/1

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

r-burns commented:
These flaky AWS tests are making me suspicious. Hydra failure:

```
/tmp/nix-build-aws-sdk-cpp-1.8.130.drv-0/source/aws-cpp-sdk-core-tests/aws/client/AWSClientTest.cpp:584: Failure
Expected equality of these values:
  Aws::Region::US_WEST_2
    Which is: "us-west-2"
  config.region
    Which is: "us-east-1"
/tmp/nix-build-aws-sdk-cpp-1.8.130.drv-0/source/aws-cpp-sdk-core-tests/aws/client/AWSClientTest.cpp:585: Failure
Expected equality of these values:
  "Dijkstra"
  config.profileName.c_str()
    Which is: "default"
```

This looks like a data race between:
https://github.com/aws/aws-sdk-cpp/blob/1.8.130/aws-cpp-sdk-core-tests/aws/client/AWSClientTest.cpp#L577
https://github.com/aws/aws-sdk-cpp/blob/1.8.130/aws-cpp-sdk-core-tests/aws/client/AWSClientTest.cpp#L744

What do you think? Maybe we should file an issue upstream and disable parallel testing instead. (The tests ran in 4 seconds on Hydra, so even in serial they should still be quick.)

happysalada committed:
aws-sdk-cpp: disable parallel checking

some data races are suspected in
aws-cpp-sdk-core-tests/aws/auth/AWSCredentialsProviderTest.cpp
and
aws-cpp-sdk-core-tests/aws/client/AWSClientTest.cpp

happysalada commented:
I think you are right here.
I've tried disableParallellCheck. I think it's worth trying. If we see another occurence of the bug we can always think again.
I've created a clean version of that fix in https://github.com/NixOS/nixpkgs/pull/136645
(I'm not sure why github is not happy with this commit).
